e2e/types/types.go

```diff
@@ -71,6 +73,10 @@ type Workload interface {
 	// GetLabelSelector returns the labelSelector to used for selecting the resources.
 	// This value is made use during preparation of hooks used in recipe.
 	GetLabelSelector() *metav1.LabelSelector
+	// GetCheckHook returns the check hook to be executed based on workload and given namespace.
+	GetCheckHook(string) *recipe.Hook
```

Comment:
If we go with returning the entire hook, we need to remove the GetLabelSelector and GetSelectResource, because they are now internal implementation detail of Get*Hook.

But I feel that the entire hook contains too much details that the workload should not care about. It should care only about the label selector, resource string, check expression, and the command to use for execution. The rest (how to build the entire hook belong to the recipe package.